src/backend/marsha/core/xapi.py

```diff
@@ -94,14 +151,88 @@ def __init__(self, document, statement, jwt_token):
             "objectType": "Activity",
         }
 
-        self.statement = statement
+        return statement
 
 
 class XAPIVideoStatement(XAPIStatementMixin):
     """Object managing statement for video objects."""
 
-    def __init__(self, video, statement, jwt_token):
-        """Compute a valid xapi satement.
+    def from_website(self, video, statement, current_site, user):
+        """Compute a valid xapi satement in a website context.
+
+        Parameters
+        ----------
+        video : Type[.models/videos]
+            The video object used in the xAPI statement
+
+        statement : dictionary
+            Statement containing base information to send to the LRS
+            An example of expected statement:
+            {
+                "verb": {
+                    "id": "http://adlnet.gov/expapi/verbs/initialized",
+                    "display": {
+                        "en-US": "initialized"
+                    }
+                },
+                "context": {
+                    "extensions": {
+                        "https://w3id.org/xapi/video/extensions/volume": 1,
+                        "https://w3id.org/xapi/video/extensions/video-playback-size": "640x264",
+                    }
+                }
+            }
+
+        current_site : Type[django.contrib.sites.models.Site]
+            The current site used to send the XAPI request
+
+        user: Type[marsha.core.models.User]
+            The connected user who sent the XAPI request
+
+        """
+        homepage = current_site.domain
+
+        if re.match(r"^http(s?):\/\/.*", homepage) is None:
+            homepage = f"http://{homepage}"
+
+        activity_type = "https://w3id.org/xapi/video/activity-type/video"
+
+        # When the video is a live we change the activity to webinar
+        if video.is_live:
+            activity_type = "http://id.tincanapi.com/activitytype/webinar"
+
+        if "id" not in statement:
+            statement["id"] = str(uuid.uuid4())
+
+        statement["timestamp"] = timezone.now().isoformat()
```

Comment:
Maybe we could factorize the main part of the statement build in the `XAPIStatementMixin` ?